Packages/ClientRuntime/Tests/ClientRuntimeTests/PrimitiveTypeExtensionsTests/StringExtensionsTests.swift

```diff
@@ -96,4 +96,40 @@ class StringExtensionsTests: XCTestCase {
             XCTAssertEqual(string.substringBefore(":"), match)
         }
     }
+
+    func testOffset() {
+        let string = "FooBar"
+        let offset = string[3]
+        XCTAssertEqual("B", offset)
+    }
+
+    func testRange() {
+        let string = "FooBar"
+        let range = string[1...3]
+        XCTAssertEqual("ooB", range)
+    }
+
+    func testClosedRange() {
+        let string = "FooBar"
+        let range = string[1...]
+        XCTAssertEqual("ooBar", range)
+    }
+
+    func testPartialRangeThrough() {
+        let string = "FooBar"
+        let range = string[...3]
+        XCTAssertEqual("FooB", range)
+    }
+
+    func testPartialRangeUpTo() {
+        let string = "FooBar"
+        let range = string[..<3]
+        XCTAssertEqual("Foo", range)
+    }
+
+    func testPartialRangeFrom() {
+        let string = "FooBar"
+        let range = string[3...]
+        XCTAssertEqual("Bar", range)
+    }
```

Comment:
Since emojis are valid identifiers for Swift, you should add tests for emojis here too. I think your subscript extensions won't work for any characters that aren't UTF-8

Reply:
AFIK, HTTP headers are only ASCII but some services may end up doing some crazy things. 

That said, now it is all `Index` based.